apps/trackers/jira/query.py

```diff
@@ -128,6 +128,16 @@ def generate_labels(self):
             )
         )
 
+        # If all affects are NEW and if this tracker is New, add label validation-requested.
+        # Status values governed by Jira project workflow, not accessible through JiraProjectFields.
+        # Most probably, New will always be something like NEW, New, new.
+        if self.tracker.status.lower() == "new":
```

Comment:
I am not sure this condition should be there. I think we do not care for the tracker status very much. That is rather engineering think. Mostly the label will only be added on creation and as we do not set the tracker status I assume it will become some kind of **new** by default. Theoretically if someone removes the label while from our point of view it should still be there it would be re-added - which I think should happen regardless of the tracker status. I do no see any other option. WDYT?

Reply:
> I am not sure this condition should be there. I think we do not care for the tracker status very much. That is rather engineering think. Mostly the label will only be added on creation and as we do not set the tracker status I assume it will become some kind of new by default.

Changed the code in today's commit to only ever add the label when the tracker is being created, i.e. if the tracker already exists and moves to New from any other state, the label is not added. Re-reading the Description of the story, it seems today's commit brings the PR to what has been asked in Description (sorry for the confusion).

> Theoretically if someone removes the label while from our point of view it should still be there it would be re-added - which I think should happen regardless of the tracker status. I do no see any other option. WDYT?

I agree on no other options existing, but probably disagree on what these options are.

With today's commit, the label would **not** be regenerated in such a scenario. Before today's commit, it would be regenerated. Which do you want, then? Without today's commit or with it? I don't see a viable algorithm that would fulfill all requirements in any other way (ignoring tracker state but re-adding the label if state is New is logically inconsistent, unless I'm missing something).

Reply:
Can we merge this PR? I think this discussion became somewhat out-of-date. If someone doesn't like the current behavior (the label is never re-added, once removed), they can file a bug.

Reply:
Thank you! We seem to agree on leaving it as it is now then.

(for posterity, the messages from Jan 18 and Jan 22 happened in parallel universes without knowing about each other, then the universes merged today)